#!/usr/bin/env python
PACKAGE = "vision"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("low_h_param",    int_t,    0, "An Integer parameter", 0,  0, 180)
gen.add("low_s_param",    int_t,    0, "An Integer parameter", 0,  0, 255)
gen.add("low_v_param",    int_t,    0, "An Integer parameter", 0,  0, 255)

gen.add("high_h_param",    int_t,    0, "An Integer parameter", 180,  0, 180)
gen.add("high_s_param",    int_t,    0, "An Integer parameter", 255,  0, 255)
gen.add("high_v_param",    int_t,    0, "An Integer parameter", 255,  0, 255)

gen.add("epsilon_param",   int_t,    0, "An Integer parameter", 5, 0, 20)

# gen.add("double_param", double_t, 0, "A double parameter",    .5, 0,   1)
# gen.add("str_param",    str_t,    0, "A string parameter",  "Hello World")
# gen.add("bool_param",   bool_t,   0, "A Boolean parameter",  True)

# size_enum = gen.enum([ gen.const("Small",      int_t, 0, "A small constant"),
#                        gen.const("Medium",     int_t, 1, "A medium constant"),
#                        gen.const("Large",      int_t, 2, "A large constant"),
#                        gen.const("ExtraLarge", int_t, 3, "An extra large constant")],
#                      "An enum to set size")

# gen.add("size", int_t, 0, "A size parameter which is edited via an enum", 1, 0, 3, edit_method=size_enum)

exit(gen.generate(PACKAGE, "vision_node", "VisionNode"))